test/common/upstream/least_request_load_balancer_fuzz_test.cc

```diff
@@ -0,0 +1,83 @@
+#include <memory>
+
+#include "test/common/upstream/least_request_load_balancer_fuzz.pb.validate.h"
+#include "test/common/upstream/zone_aware_load_balancer_fuzz_base.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+// Least Request takes into account both weights (handled in ZoneAwareLoadBalancerFuzzBase), and
+// requests active as well
+void setRequestsActiveForStaticHosts(NiceMock<MockPrioritySet>& priority_set,
+                                     const std::string& random_bytestring) {
+  uint32_t index_of_random_bytestring = 0;
+  // Iterate through all the current host sets and set requests for each
+  for (uint32_t priority_level = 0; priority_level < priority_set.hostSetsPerPriority().size();
+       ++priority_level) {
+    MockHostSet& host_set = *priority_set.getMockHostSet(priority_level);
+    for (auto& host : host_set.hosts_) {
+      // Make sure no weights persisted from previous fuzz iterations
+      ASSERT(host->stats().rq_active_.value() == 0);
+      host->stats().rq_active_.set(
+          random_bytestring[index_of_random_bytestring % random_bytestring.length()] % 3);
+      ++index_of_random_bytestring;
+    }
+  }
+}
+
+void removeRequestsActiveForStaticHosts(NiceMock<MockPrioritySet>& priority_set) {
+  // Clear out any set requests active
+  for (uint32_t priority_level = 0; priority_level < priority_set.hostSetsPerPriority().size();
+       ++priority_level) {
+    MockHostSet& host_set = *priority_set.getMockHostSet(priority_level);
+    for (auto& host : host_set.hosts_) {
+      host->stats().rq_active_.set(0);
+    }
+  }
+}
+
+DEFINE_PROTO_FUZZER(const test::common::upstream::LeastRequestLoadBalancerTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  }
+
+  const test::common::upstream::ZoneAwareLoadBalancerTestCase& zone_aware_load_balancer_test_case =
+      input.zone_aware_load_balancer_test_case();
+
+  ZoneAwareLoadBalancerFuzzBase zone_aware_load_balancer_fuzz = ZoneAwareLoadBalancerFuzzBase(
+      zone_aware_load_balancer_test_case.need_local_priority_set(),
+      zone_aware_load_balancer_test_case.random_bytestring_for_weights());
+  zone_aware_load_balancer_fuzz.initializeLbComponents(
+      zone_aware_load_balancer_test_case.load_balancer_test_case());
+
+  setRequestsActiveForStaticHosts(zone_aware_load_balancer_fuzz.priority_set_,
```

Comment:
why doesn't active requests change over time?

Reply:
To document, this is what we talked about in standup. Let me brainstorm some ideas about being able to decorate action streams with more specific proto fields.

Reply:
I think since we already have a static initialization of things currently checked into master, let's make scope of this pr to just get least request out there. I'll spend the rest of the day thinking about a design that allows decorated logic for action streams.

Reply:
Just a thought -- you could add another random byte string to `UpdateHealthFlags update_health_flags = 1;`
and more adaptly name it "UpdatePrioritySet", which updates both health flags, and if needed, active requests for that priority set.

Reply:
Yeah this is a great idea! However, I think since this pr has been sitting here for a while, let's get this merged first :). I'll keep this design decision in mind as I clean up Load Balancing Fuzzing/and or add Maglev/RingHash (I saw some weight logic in both unit tests).